### ISSUE TITLE
I/O Streaming: way to wait for write stream to drain

### ISSUE THREAD

**Issue by ambischof** [exceljs/exceljs#464]
tldr: for the workbookWriter, the row commit should return true or false as to whether the write stream is ready to accept more data.

I am using the I/O streaming method

```

var workbook = new exceljs.stream.xlsx.WorkbookWriter({stream: stream});
var worksheet = workbook.addWorksheet();

...

while ( !row.done) {
	row = resultSetIterator.rows.next();
	worksheet.addRow(row.value).commit();
}
...
worksheet.commit();
workbook.commit().then(resolve);
```

In my use case, my input of row is just an iterator which means that the input will come in far faster than the output can be processed, and because it is a while loop, it is basically blocking. Typically for streams, when you write to them, it returns true or false as to whether the write stream will accept any more data, but `commit` returns nothing.

Because the result set can be arbitrarily large, it is not practical to work in a non-streaming fashion.

Is there already some method to see if the stream needs draining? Without the ability to see if the write stream needs to drain, it completely misses the point of streaming at all.

(FWIW, to answer my own question, yes there is one already, `worksheet.stream.pipes[0]._writableState.needDrain` but I'd like a method that is less hackish).

If something already exists that I'm just missing, please tell me.

**Comment by chianquan** [comment]
The same case.
`worksheet.stream.pipes[0]._writableState.needDrain` is useful ,but wish  less hackish.
